File: csharp/webApiPro/SqlModel/Flight/MainDelayReasonRate.cs
namespace webApiPro.SqlModel.Flight
{
    public class MainDelayReasonRate
    {
        public int Year { get; set; }
        public int Month { get; set; }
        public int NumCarrierDelays {  get; set; }
        public int NumWeatherDelays { get; set; }
        public int NumNASDelays { get; set; }
        public int NumSecurityDelays { get; set; }
        public int NumLateAircraftDelays { get; set; }
        public int TotalDelays { get; set; }
        public double CarrierDelayRate { get; set; }
        public double WeatherDelayRate { get; set;}
        public double NASDelayRate { get; set; }
        public double SecurityDelayRate { get; set; }
        public double LateAircraftDelayRate { get;set; }
        
    }
}

File: spring_boot/springWeb/src/main/java/com/mhdx/springweb/model/Sql/MainDelayReasonRate.java
package com.mhdx.springweb.model.Sql;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;

public class MainDelayReasonRate {
    @Getter
    private Integer Year;
    @Getter
    private Integer Month;
    @Getter
    private Integer numCarrierDelays;
    @Getter
    private Integer numWeatherDelays;
    @Getter
    private Integer numNASDelays;
    @Getter
    private Integer numSecurityDelays;
    @Getter
    private Integer numLateAircraftDelays;
    @Getter
    private Integer totalDelays;
    @Getter
    private Double carrierDelayRate;
    @Getter
    private Double weatherDelayRate;
    @Getter
    private Double nASDelayRate;
    @Getter
    private Double securityDelayRate;
    @Getter
    private Double lateAircraftDelayRate;

    public MainDelayReasonRate(Integer Year, Integer Month, Integer numCarrierDelays, Integer numWeatherDelays,
                               Integer numNASDelays, Integer numSecurityDelays, Integer numLateAircraftDelays, Integer totalDelays,
                               Double carrierDelayRate, Double weatherDelayRate, Double nASDelayRate, Double securityDelayRate,
                               Double lateAircraftDelayRate) {
        this.Year = Year;
        this.Month = Month;
        this.numCarrierDelays = numCarrierDelays;
        this.numWeatherDelays = numWeatherDelays;
        this.numNASDelays = numNASDelays;
        this.numSecurityDelays = numSecurityDelays;
        this.numLateAircraftDelays = numLateAircraftDelays;
        this.totalDelays = totalDelays;
        this.carrierDelayRate = carrierDelayRate;
        this.weatherDelayRate = weatherDelayRate;
        this.nASDelayRate = nASDelayRate;
        this.securityDelayRate = securityDelayRate;
        this.lateAircraftDelayRate = lateAircraftDelayRate;
    }
}

